apps/api/src/controllers/request-helpers.ts

```diff
@@ -160,3 +162,53 @@ export function parseAdminSearchQuery(
   }
   return data;
 }
+
+/**
+ * Check if the request is an unauthenticated *or* user authenticated (frontend) request
+ * @param req Express request
+ * @returns True if the request is authenticated as an admin, false otherwise
+ */
+export function checkIsPublicOrUserRequest(req: Request) {
+  return !req.agentId && !checkIsAdmin(req);
+}
+
+/**
+ * Check if the cache is enabled
+ * @returns True if the cache is enabled, false otherwise
+ */
+export function checkIsCacheEnabled() {
+  return !config.cache.api.disableCaching;
+}
+
+/**
+ * Check if the cache should be created for the request
+ * @param req Express request
+ * @returns True if the cache should be created, false otherwise
+ */
+export function checkShouldCacheResponse(req: Request) {
```

Comment:
we cache anything from an unauthenticated or authenticated user. agent / API key requests will get the freshest data.